osidb/models.py

```diff
@@ -796,6 +796,36 @@ def _validate_cwe_format(self):
             if not re.match(r"^(CWE-[0-9]+|\(CWE-[0-9]+(\|CWE-[0-9]+)*\))$", element):
                 raise ValidationError("CWE IDs is not well formated.")
 
+    def _validate_unsupported_impact_change(self):
+        """
+        Check that an update of a flaw with open trackers does not change between
+        Critical/Important/Major Incident and Moderate/Low.
+        """
+        if self._state.adding:
+            return
+
+        old_flaw = Flaw.objects.get(pk=self.pk)
+        was_high_impact = (
+            old_flaw.impact in [FlawImpact.CRITICAL, FlawImpact.IMPORTANT]
+            or old_flaw.is_major_incident
+        )
+        still_high_impact = (
+            self.impact in [FlawImpact.CRITICAL, FlawImpact.IMPORTANT]
+            or self.is_major_incident
+        )
+        if (
+            was_high_impact
+            and not still_high_impact
+            and Tracker.objects.filter(affects__flaw=self)
+            .exclude(status="Closed")
+            .exists()
+        ):
+            raise ValidationError(
```

Comment:
In the original SFM2 validation you can see that it starts by `@register.warning` which means that it is not a blocking error but rather a notification to the user that something potentially bad happened. The reason is that sometimes it is necessary to change the impact in an unsupported (by tooling) way. The user is then responsible for cleaning everything accordingly just manually. Instead of `raise` there should be just `self.alert(...`

Reply:
Makes sense, will send fixes.